### ISSUE TITLE
Implement the `mup-Depth`

### ISSUE THREAD

**Issue by DomInvivo** [datamol-io/graphium#499]
Newer paper about depth scaling: [Tensor Programs VI: Feature Learning in Infinite-Depth Neural Networks](https://arxiv.org/abs/2310.02244). 

If my understanding is correct, this is just about these 2 steps

1. Multiplying the skip connection by a factor `a / √L`, which can be accomplished easily with our `ResidualConnectionBase` class.
2. Multiplying the learning rate by a factor `n` or `n / √L` depending on the optimizer. Not sure if they have a new repo for this.

![image](https://github.com/datamol-io/graphium/assets/47570400/22743509-a3bd-4cfb-8793-0d82c1cd20fe)
